app/assets/javascripts/checkouts.js

```diff
@@ -1,21 +1,65 @@
 $(document).ready(function(){
 
+    // Make a call to /return when a new barcode is entered.
+    $('#g-barcode').change(function(){
+        var barcode_val = $(this).val();
+
+        if(!bc_regex.test(barcode_val)){
+            $.notify('Invalid barcode format! Barcode should be at least 3 characters long and only contain alphanumeric characters.', 'warning', 5000);
+            $(this).val('');
+            return;
+        }
+        gameBarcode(false);
+
+        $.post('/return', { barcode: barcode_val }).success(function(response){
+            if(response.errors){
+                $.each(response.errors, function(k, v){
+                    $.notify(v, 'danger');
+                });
+                gameBarcode(true);
+            }else if(response.time){
+                $.notify('Returned game successfully!');
```

Comment:
Is it much work to include the game title on a successful return? It would help to see that and maybe extend the length of this notification so that people can double check what they just did if they're scanning a game and not paying attention or scanning more than one game.